plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -73,27 +72,38 @@
 import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
 
 @TestInstance(PER_CLASS)
-public class BaseTestHiveOnDataLake
+abstract class BaseTestHiveOnDataLake
         extends AbstractTestQueryFramework
 {
-    private static final String HIVE_TEST_SCHEMA = "hive_datalake";
-    private static final DataSize HIVE_S3_STREAMING_PART_SIZE = DataSize.of(5, MEGABYTE);
+    protected static final String HIVE_TEST_SCHEMA = "hive_datalake";
+    protected static final DataSize HIVE_S3_STREAMING_PART_SIZE = DataSize.of(5, MEGABYTE);
 
-    private String bucketName;
-    private Hive3MinioDataLake hiveMinioDataLake;
-    private HiveMetastore metastoreClient;
+    protected String bucketName;
+    protected HiveMinioDataLake hiveMinioDataLake;
+    protected HiveMetastore metastoreClient;
+
+    @BeforeAll
+    public void setUp()
+    {
+        computeActual(format(
+                "CREATE SCHEMA hive.%1$s WITH (location='s3a://%2$s/%1$s')",
+                HIVE_TEST_SCHEMA,
+                bucketName));
+        computeActual("CREATE SCHEMA hive.functions");
+    }
+
+    protected abstract HiveMinioDataLake hiveMinioDataLake();
```

Comment:
We could add a constructor that accepts `HiveMinioDataLake` instead of adding a protected method.

Reply:
added a constructor.

Reply:
test failed with https://github.com/trinodb/trino/actions/runs/12402571060/job/34624391815?pr=24418#step:5:2540 which had failed with the same error before as well. I'll revert this to use abstract method.

Reply:
The error means we shouldn't reuse the same bucket name among tests. You can see a comment in `MinioClient#makeBucket`. The cause is you changed `bucketName` to static.

Reply:
ya I saw that but wasn't able to reproduce it. I think it could have been reproduced if I would have run in parallel.I moved bucket name to sub classes now.